tests/stdlib/tos.nim

```diff
@@ -26,6 +26,31 @@ Raises
 # test os path creation, iteration, and deletion
 
 import os, strutils, pathnorm
+import "$nim/compiler/unittest_light"
```

Comment:
unittest is bad because it's full of ``when compiles``. Guess what, unittest_light repeats the mistake. The tests were fine before without this import.

Reply:
I removed that import for the sake of moving fwd with this PR, but please explain what is wrong with `unittest_light` having `compiles` ?
* assertEquals is written as `if lhs!=rhs: doAssert false, mismatch(lhs, rhs)` so the `compiles` is just used for printing the error msg (inside `mismatch`), not for correctness validation
* theres is already a test for that tests/compiler/tunittest_light.nim which tests `assertEquals` using `doAssert` so if `assertEquals` had bugs, `doAssert` would catch it

If it's a compilation speed issue, there are easy options.

More generally, for `assertEquals`, `unittest.check`, `fmt`, and `sequtils`: 1 argument I heard against more widespread use in testament was "who is guarding the guardian" argument; I'm all for adding more thorough testing (as needed) for these but we shouldn't use `FUD` arguments that they may have bugs that would render testament useless; 
If a bug ever slips through existing checks, it'll eventually be found in the wild, reported, and fixed;
note, outside of testament, these are pretty popular tools for testing in nimble packages, for a reason.

Reply:
nim-regex doesn't compile with devel because it uses unittest and unittest does nasty stuff. Yeah, these are compiler regressions, but still, a testing infrastructure shouldn't stress the compiler so much that it becomes hard to debug itself. No FUD here, just lots of experience with unittest.

Reply:
the concerns you're raising seem specific to `unittest`, not `unittest_light`, as I described here https://github.com/nim-lang/Nim/pull/10545#discussion_r254069436